src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -85,14 +87,13 @@ public FileUploadSubmissionResource(SubmissionRepository submissionRepository, R
 
         final var exercise = fileUploadExerciseRepository.findOneByIdElseThrow(exerciseId);
         final User user = userRepository.getUserWithGroupsAndAuthorities();
-        if (!authCheckService.isAtLeastStudentForExercise(exercise, user)) {
-            return forbidden();
-        }
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.STUDENT, exercise, user);
 
-        // Make sure that the exercise exists
-        if (exercise == null) {
-            return ResponseEntity.badRequest()
-                    .headers(HeaderUtil.createFailureAlert(applicationName, true, "submission", "exerciseNotFound", "No exercise was found for the given ID.")).body(null);
+        // if there is a participation that has an exercise linked to it,
+        // the exercise needs to be the same as the one referenced in the path via exerciseId
+        if (fileUploadSubmission.getParticipation() != null && fileUploadSubmission.getParticipation().getExercise() != null
+                && !fileUploadSubmission.getParticipation().getExercise().getId().equals(exerciseId)) {
```

Comment:
What will now happen if the exercise does not exist for the given ID (illegal request) OR if the request body does not have a participation?

Reply:
In the case of the exerciseId not having a matching exercise, a "404 Not Found" will be returned, as is usual for the "findByIdElseThrow" methods. As was the case before, nothing out of the ordinary happens. This just ensures that IF there is a link to an exercise in the body, it has to be the same as referenced in the path, otherwise, a "400 Bad Request" gets returned.